sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/action/permission/add/AddPermissionRequest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.rest.action.permission.add;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+public class AddPermissionRequest extends ActionRequest implements ToXContentObject {
+
+    private String permissionString;
+
+    private String principalString;
+
+    public AddPermissionRequest(StreamInput in) throws IOException {
+        super(in);
+        this.permissionString = in.readString();
+        this.principalString = in.readString();
+    }
+
+    public AddPermissionRequest(String permissionString, String principalString) {
+        this.permissionString = permissionString;
+        this.principalString = principalString;
+    }
+
+    public AddPermissionRequest() {}
+
+    public String getPermissionString() {
+        return this.permissionString;
+    }
+
+    public void setPermissionString(String permissionString) {
+        this.permissionString = permissionString;
+    }
+
+    public String getPrincipalString() {
+        return this.principalString;
+    }
+
+    public void setPrincipalString(String principalString) {
+        this.principalString = principalString;
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException validationException = null;
+        if (permissionString == null) { // TODO: check the condition once
+            validationException = addValidationError("No permissionString specified", validationException);
+        } else if (principalString == null) {
+            validationException = addValidationError("No principal specified", validationException);
+        }
+        return validationException;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(permissionString);
+        out.writeString(principalString);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject();
+        builder.value(permissionString);
+        builder.value(principalString);
+        builder.endObject();
+        return builder;
+    }
+
+    public void fromXContent(XContentParser parser) throws IOException { // TODO: Talk to DC about this
```

Comment:
What did DC say? :D 

To me this feels pretty clunky for two fields, lots of duplicated logic that could lead to copypasta errors.  I believe there are alternatives that I haven't used, I'd be curious what your take is.

Reply:
ConstructingObjectParser seems much cleaner IMO

Reply:
I think the thought was to stick with this until we decided to move away from cores conventions. That being said, DC did implement the Object Parser version and I did as a result too since I based my work off of his. So it should be as quick as switching the connections down the road. I agree it is not the prettiest code but I also get that DC wants to stick with core's convention. (I could also be misremembering...)

Reply:
It looks like core has two conventions, so we get to pick either one

Reply:
Oh gothca. I will check in with DC then and him and I will agree on one to pick and standardize for Identity. @DarshitChanpura do you have any thoughts on which one you prefer since core uses both conventions? Sorry if I misunderstood you before as well. I don't want to speak for you incorrectly.